cmd/internal/kube/daemonset/daemonset.go

```diff
@@ -95,8 +98,41 @@ func (d *Daemonset) KubernetesEventsWorker() {
 	// Not implemented
 }
 
-func (d *Daemonset) PodsHealthWorker() {
-	// Not implemented
+func (d *Daemonset) checkPodsHealth() {
+	podNames := d.PodNameStopChanMap.Keys()
+
+	podStatuses, err := d.KubeClient.GetPodsStatus(podNames)
+	if err != nil {
+		printer.Errorf("failed to get pods status: %v", err)
+		return
+	}
+
+	for podName, podStatus := range podStatuses {
+		if podStatus == string(coreV1.PodSucceeded) || podStatus == string(coreV1.PodFailed) {
+			printer.Infof("pod %s has stopped running", podStatus)
+			d.StopApiDumpProcess(
```

Comment:
This adds an additional case where concurrency may occur in StopApiDumpProcess; we may learn about a stopped pod nearly-simultaneously from this worker and from the event channel.  I think that is manageable, but it has to be considered.